qiita_ware/ebi.py

```diff
@@ -680,23 +704,70 @@ def parse_EBI_reply(self, curl_result):
 
         Returns
         -------
-        study_accession
-            The study accession number, in case of failure it returns None
-        submission_accession
-            The submission accession number, in case of failure it returns None
+        bool
+            If the submission was successful or not
+        study_accession : str
+            The study accession number. None in case of failure
+        sample_accessions : dict of {str: str}
+            The sample accession numbers, keyed by sample id. None in case of
+            failure
+        biosample_accessions : dict of {str: str}
+            The biosample accession numbers, keyed by sample id. None in case
+            of failure
+        experiment_accessions : dict of {str: str}
+            The experiment accession numbers, keyed by sample id. None in case
+            of failure
+        run_accessions : dict of {str: str}
+            The run accession numbers, keyed by sample id. None in case of
+            failure
+
+        Raises
+        ------
+        EBISubmissionError
+            If curl_result is not a valid XML file
+            If the ebi subumission has not been successful
+            If multiple study tags are found in the curl result
         """
-        study_accession = None
-        submission_accession = None
-
-        if 'success="true"' in curl_result:
-            accessions = search('<STUDY accession="(?P<study>.+?)".*?'
-                                '<SUBMISSION accession="(?P<submission>.+?)"',
-                                curl_result)
-            if accessions is not None:
-                study_accession = accessions.group('study')
-                submission_accession = accessions.group('submission')
-
-        return study_accession, submission_accession
+        try:
+            root = ET.fromstring(curl_result)
+        except ParseError:
+            raise EBISubmissionError(
+                "The curl result from the EBI submission doesn't look like "
+                "an XML file:\n%s" % curl_result)
```

Comment:
AFAIK, EBISubmissions can currently be done only by admin users. Anyways, I think is a good idea on moving this to the log table and just point the log id in the raised error.